UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -1105,23 +1097,15 @@ public enum WeaponType
 		Burst = 2
 	}
 
-#if UNITY_EDITOR
-	[CustomEditor(typeof(Gun), true)]
-	public class GunEditor : Editor
+	/// <summary>
+	///	//TODO: write a half decent summary
+	/// </summary>
+	public enum TriggerBehaviour
```

Comment:
this enum represents particular classes, doesn't it? I think we should test using the type itself intead of enum values you have to update every time you add a new strategy.

Reply:
the only reason this enum exists is because returning a number isn't particularly clear as to what the fuck it does

Reply:
oh, the strategies are just numbers?

Reply:
a method in a component on the firing pin is called, the method returns a value which effects what will happen, be it nothing, working normally or playing the bike horn sfx

Reply:
why don't we make the pin act by itself instead of telling weapon how to act?

Reply:
i assume you mean either calling a method on the guns component, which would require creating a method that only contains the switch case or alternatively moving over all firing behaviour to the pin itself, which is a huge amount of work

Reply:
I think the later is the way to go. Instead of switching what the pin returns, you have a base abstract pin class and make the rest inherit from it. Add an abstract trigger method and call that from Gun.cs with a reference to the abstract class so it works whatever pin you have attached.

Reply:
I understand all of this is a lot of work. This PR intends to clean these scripts and honestly one of the biggest problems with Gun.cs is that it does way too many things. If you're not willing to do that level of clean up, it is aight, we can do baby steps until that class nice and tidy.

Reply:
i dont see how it would be possible to even seperate them into seperate components or something
the if2 interactions are 1/4th of the script alone

Reply:
well, the pin behaviours sound like a good start. You would still be calling them from the IF2 methods inside Gun.cs but what pulling the triggers does will be called directly from pin object with the structure I talked about above. If there is something I'm not seeing, let me know.